keyboards/gray_studio/think65/solder/keymaps/yt/keymap.c

```diff
@@ -0,0 +1,148 @@
+/* Copyright 2021 Qiaowei Tang
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+ /*
+ * LED ranges for Think6.5v2 2U
+ * These values were derived from manual testing. Derived from keymaps/rys.
+ * ┌───────┬───────┬─────────────┬───────────────────────────────────────────┐
+ * │ 00 01 │ 02 03 │ 04 05 06 07 │ 08 09 10 11 12 13 14 15 16 17 18 19 20 21 │
+ * │ badge │ badge │    (?)      │              underglow (?)                │
+ * │  bar  │ icon  │             │                                           │
+ * └───────┴───────┴─────────────┴───────────────────────────────────────────┘
+ */
+
+
+// Setup some layers to control mac/win layouts
+#include QMK_KEYBOARD_H
+
+enum bs_layers {
+    _WIN,
+    _MAC,
+    _FN,
+    _BLANC
+};
+
+// Setup some keycodes to control cycling and range toggling
+enum bs_keycodes {
+  WIN = SAFE_RANGE,
+  MAC
+};
+
+// Light LEDs 6 to 9 and 12 to 15 red when caps lock is active. Hard to ignore!
+const rgblight_segment_t PROGMEM my_capslock_layer[] = RGBLIGHT_LAYER_SEGMENTS(
+    {0, 4, HSV_WHITE}     // Light 4 LEDs, starting with LED 12
+);
+
+// Now define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    my_capslock_layer     // Overrides other layers
+);
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = my_rgb_layers;
+};
+
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    return true;
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case MAC:
+            if (record->event.pressed) {
+                rgblight_mode(6);
```

Comment:
2 things here: 

You should probably be using `rgblight_mode_noeeprom` so you're not writing to the eeprom. 

Also, you can use `default_layer_state_set_user` to trigger the modes.  Eg: 

```c
layer_state_t default_layer_state_set_user(layer_state_t state) {
    if (get_highest_layer(state) == _MAC) {
        rgblight_mode_noeeprom(6);
    } else {
        rgblight_mode_noeeprom(9);
    }
    return state;
}
```

Reply:
I've updated this change. and tested

Reply:
I've removed the disable leader. Let me know if there is anything additional u want changed.